### PR TITLE
Remove subst_list from ert_workflow_list

### PR DESCRIPTION
**Pull request by DanSava**
**Issue**
Related to  # #4209

## Pre review checklist

- [ ] Added appropriate release note label
- [ ] PR title captures the intent of the changes, and is fitting for release notes.
- [ ] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

Adding labels helps the maintainers when writing release notes. This is the [list of release note labels](https://github.com/equinor/ert/labels?q=release-notes).

### CODE REVIEW

**Comment by DanSava** [comment]
Jenkins test this, please!

File: src/ert/cli/workflow.py [hunk 1]
```diff
@@ -10,7 +10,7 @@ def execute_workflow(ert, workflow_name):
         msg = "Workflow {} is not in the list of available workflows"
         logger.error(msg.format(workflow_name))
         return
-    context = workflow_list.getContext()
+    context = ert.getDataKW()
```
Comment (oysteoh) [inline src/ert/cli/workflow.py:13]: is `getDataKW()` equivalent to `getContext()` ?

Comment (oysteoh) [reply src/ert/cli/workflow.py:13]: yea, it seems like that when i'm poking around. Would it be a disaster actually renaming this to something sane? Or do we end up breaking some api? Mixing context, getDataKW and SubstitutionList here - not making any sense for me the names....  Might just be only me confused though, if so - nevermind - i can live with it :sweat_smile:

Comment (DanSava) [reply src/ert/cli/workflow.py]: Seems that renaming `getDataKW` to `get_context` does not break any API that I could find and will clear some of the name overloading for the SubstitutionList

**Comment by DanSava** [comment]
Jenkins test this, please!

**Comment by DanSava** [comment]
Jenkins test this, please!

**Comment by DanSava** [comment]
Jenkins test this, please!

**Review by oysteoh — APPROVED** [review]

**Pull request merged**